File: 2019/day01/puzzle01-1.rb
def main
  module_masses = File.readlines("input.txt").map { |line| line.strip.to_i }
  fuels = module_masses.map { |mass| fuel_required_for mass }
  total_fuel = fuels.reduce :+
  puts total_fuel
end

def fuel_required_for(mass)
  mass / 3 - 2
end

main

File: 2019/day01/puzzle01-1.py
from functools import reduce


def main():
    module_masses = read_masses("input.txt")
    fuels = [fuel_required_for(mass) for mass in module_masses]
    total_fuels = reduce(lambda a, b: a+b, fuels, 0)
    print(total_fuels)


def read_masses(filename):
    with open(filename, 'r') as file:
        masses = file.read().splitlines()
        return [int(mass) for mass in masses]


def fuel_required_for(mass):
    return mass // 3 - 2


if __name__ == "__main__":
    main()
